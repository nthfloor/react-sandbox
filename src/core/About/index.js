import React from 'react'

const About = () => (
  <div className="Home">
    <p>About</p>
  </div>
)

export default About
