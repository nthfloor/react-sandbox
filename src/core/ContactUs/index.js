import React from 'react'

const ContactUs = () => (
  <div className="ContactUs">
    <p>Contact Us</p>
  </div>
)

export default ContactUs
