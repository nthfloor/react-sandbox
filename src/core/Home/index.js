import React from 'react'
// import ShoppingList from 'core/ShoppingList'
import Game from 'core/Game'

const Home = () => (
  <div className="Home">
    <h1>This is the Home page</h1>
    
    {/* <ShoppingList name="Nathan" /> */}
    <Game />
  </div>
)

export default Home
