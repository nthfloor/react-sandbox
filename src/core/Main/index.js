import React from 'react'
import {Switch, Route} from 'react-router-dom'
import Home from 'core/Home'
import About from 'core/About'
import ContactUs from 'core/ContactUs'

import './Main.css'

const Main = () => (
  <main className="App-intro">
    <Switch>
      <Route exact path="/" component={Home} />
      <Route exact path="/about" component={About} />
      <Route exact path="/contact" component={ContactUs} />
    </Switch>
  </main>
)

export default Main
