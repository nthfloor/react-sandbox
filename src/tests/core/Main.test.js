import { shallow } from '../../enzyme';
import * as React from 'react';

import Home from 'core/Home'
import About from 'core/About'
import ContactUs from 'core/ContactUs'

import Main from 'core/Main';

describe("Main", () => {

  describe('component', () => {

    let element;

    beforeEach(() => {
      element = <Main />
    });

    it('renders as expected', () => {
      const component = shallow(element);
      expect(component).toMatchSnapshot();
    });

    it('routes / to Home', () => {
      const component = shallow(element);
      expect(component.find('Route[exact=true][path="/"]').first().prop('component')).toBe(Home);
    });

    it('routes /about to About', () => {
      const component = shallow(element);
      expect(component.find('Route[exact=true][path="/about"]').first().prop('component')).toBe(About);
    });

    it('routes /contact to ContactUs', () => {
      const component = shallow(element);
      expect(component.find('Route[path="/contact"]').first().prop('component')).toBe(ContactUs);
    });

  });

});