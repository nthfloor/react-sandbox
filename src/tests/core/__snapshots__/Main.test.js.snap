// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component renders as expected 1`] = `ShallowWrapper {}`;
